Support for PostgreSQL 18

Hello there 👋 . Im wondering if there is any plan / timeline on support PG18 now that PGBeta1 is out.

Thanks.

I'm trying to verify where `pglogical` breaks on version 18beta1. 

I followed these steps:
`cp -rvp compat17 compat18`

No changes, of course, got the following:
```
pglogical_apply.c:138:12: warning: no previous extern declaration for non-static variable 'lsn_mapping' [-Wmissing-variable-declarations]
  138 | dlist_head lsn_mapping = DLIST_STATIC_INIT(lsn_mapping);
      |            ^
pglogical_apply.c:138:1: note: declare 'static' if the variable is not intended to be used outside of this translation unit
  138 | dlist_head lsn_mapping = DLIST_STATIC_INIT(lsn_mapping);
      | ^
pglogical_apply.c:1618:8: error: too many arguments to function call, expected 6, have 7
 1615 |                 (void) PortalRun(portal, FETCH_ALL,
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1616 |                                                  isTopLevel,
      |                                                  ~~~~~~~~~~~
 1617 |                                                  receiver, receiver,
      |                                                  ~~~~~~~~~~~~~~~~~~~
 1618 |                                                  NULL);
      |                                                  ^~~~~
/usr/lib64/llvm20/bin/../../../lib/clang/20/include/__stddef_null.h:26:14: note: expanded from macro 'NULL'
   26 | #define NULL ((void*)0)
      |              ^~~~~~~~~~
/home/vinnix/Sources/Github/pglogical/compat18/pglogical_compat.h:25:60: note: expanded from macro 'PortalRun'
   25 |         PortalRun(portal, count, isTopLevel, true, dest, altdest, qc)
      |         ~~~~~~~~~                                                 ^~
/home/vinnix/pg-versions/REL_18_STABLE-20250706-1411/include/postgresql/server/tcop/pquery.h:38:13: note: 'PortalRun' declared here
   38 | extern bool PortalRun(Portal portal, long count, bool isTopLevel,
      |             ^         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   39 |                                           DestReceiver *dest, DestReceiver *altdest,
      |                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   40 |                                           QueryCompletion *qc);
      |                                           ~~~~~~~~~~~~~~~~~~~
pglogical_apply.c:155:29: warning: no previous extern declaration for non-static variable 'errcallback_arg' [-Wmissing-variable-declarations]
  155 | struct ActionErrCallbackArg errcallback_arg;
      |                             ^
pglogical_apply.c:155:1: note: declare 'static' if the variable is not intended to be used outside of this translation unit
  155 | struct ActionErrCallbackArg errcallback_arg;
      | ^
2 warnings and 1 error generated.
make: *** [<builtin>: pglogical_apply.o] Error 1
```

Looking the function signature of `REL_18_STABLE` at `src/backend/tcop/pquery.c`:

```
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  661) /*
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  662)  * PortalRun
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  663)  *               Run a portal's query or queries.
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  664)  *
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  665)  * count <= 0 is interpreted as a no-op: the destination gets started up
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  666)  * and shut down, but nothing else happens.  Also, count == FETCH_ALL is
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  667)  * interpreted as "all rows".  Note that count is ignored in multi-query
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  668)  * situations, where we always run the portal to completion.
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  669)  *
b9527e984092 (Tom Lane         2007-03-13 00:33:44 +0000  670)  * isTopLevel: true if query is being executed at backend "top level"
b9527e984092 (Tom Lane         2007-03-13 00:33:44 +0000  671)  * (that is, directly from a client command message)
b9527e984092 (Tom Lane         2007-03-13 00:33:44 +0000  672)  *
556f7b7bc18d (Tom Lane         2024-12-09 14:38:19 -0500  673)  * run_once: ignored, present only to avoid an API break in stable branches.
556f7b7bc18d (Tom Lane         2024-12-09 14:38:19 -0500  674)  *
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  675)  * dest: where to send output of primary (canSetTag) query
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  676)  *
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  677)  * altdest: where to send output of non-primary queries
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  678)  *
2f9661311b83 (Alvaro Herrera   2020-03-02 18:19:51 -0300  679)  * qc: where to store command completion status data.
2f9661311b83 (Alvaro Herrera   2020-03-02 18:19:51 -0300  680)  *               May be NULL if caller doesn't want status data.
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  681)  *
2eb4a831e5fb (Peter Eisentraut 2017-08-16 00:22:32 -0400  682)  * Returns true if the portal's execution is complete, false if it was
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  683)  * suspended due to exhaustion of the count parameter.
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  684)  */
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  685) bool
691b8d59281b (Robert Haas      2017-03-23 13:05:48 -0400  686) PortalRun(Portal portal, long count, bool isTopLevel, bool run_once,
79913910d4b5 (Tom Lane         2003-05-06 20:26:28 +0000  687)            DestReceiver *dest, DestReceiver *altdest,
2f9661311b83 (Alvaro Herrera   2020-03-02 18:19:51 -0300  688)            QueryCompletion *qc)
de28dc9a04c4 (Tom Lane         2003-05-02 20:54:36 +0000  689) {
```
I was trying to understand which parameters that might be missing, here I want to presume REL_18_STABLE is correct when using PortalRun, because it's actually working and passing my tests. 

Now, when looking at `master` branch:
```
e984092 (Tom Lane           2007-03-13 00:33:44 +0000  671)  * isTopLevel: true if query is being executed at backend "top level"
b9527e984092 (Tom Lane           2007-03-13 00:33:44 +0000  672)  * (that is, directly from a client command message)
b9527e984092 (Tom Lane           2007-03-13 00:33:44 +0000  673)  *
de28dc9a04c4 (Tom Lane           2003-05-02 20:54:36 +0000  674)  * dest: where to send output of primary (canSetTag) query
de28dc9a04c4 (Tom Lane           2003-05-02 20:54:36 +0000  675)  *
de28dc9a04c4 (Tom Lane           2003-05-02 20:54:36 +0000  676)  * altdest: where to send output of non-primary queries
de28dc9a04c4 (Tom Lane           2003-05-02 20:54:36 +0000  677)  *
2f9661311b83 (Álvaro Herrera     2020-03-02 18:19:51 -0300  678)  * qc: where to store command completion status data.
2f9661311b83 (Álvaro Herrera     2020-03-02 18:19:51 -0300  679)  *             May be NULL if caller doesn't want status data.
de28dc9a04c4 (Tom Lane           2003-05-02 20:54:36 +0000  680)  *
2eb4a831e5fb (Peter Eisentraut   2017-08-16 00:22:32 -0400  681)  * Returns true if the portal's execution is complete, false if it was
de28dc9a04c4 (Tom Lane           2003-05-02 20:54:36 +0000  682)  * suspended due to exhaustion of the count parameter.
de28dc9a04c4 (Tom Lane           2003-05-02 20:54:36 +0000  683)  */
de28dc9a04c4 (Tom Lane           2003-05-02 20:54:36 +0000  684) bool
3eea7a0c97e9 (Tom Lane           2024-12-09 14:38:19 -0500  685) PortalRun(Portal portal, long count, bool isTopLevel,
79913910d4b5 (Tom Lane           2003-05-06 20:26:28 +0000  686)                  DestReceiver *dest, DestReceiver *altdest,
2f9661311b83 (Álvaro Herrera     2020-03-02 18:19:51 -0300  687)                  QueryCompletion *qc)
de28dc9a04c4 (Tom Lane           2003-05-02 20:54:36 +0000  688) {
```


It looks like the commit `691b8d59281b` https://github.com/postgres/postgres/commit/691b8d59281b from Robert Haas is included on REL_18_STABLE (note this commit is from Mar 23, 2017) and is missing from `master`.

What are the commiters positions on this?

[update]
Taking a closer look at the commit, I can see it's included on REL_18_BETA1, so I think that will be included on REL_18_STABLE as well.

--- a/Makefile
+++ b/Makefile
@@ -69,6 +69,7 @@ EXTRA_CLEAN += compat94/pglogical_compat.o compat95/pglogical_compat.o \
 			   compat15/pglogical_compat.o compat15/pglogical_compat.bc \
 			   compat16/pglogical_compat.o compat16/pglogical_compat.bc \
 			   compat17/pglogical_compat.o compat17/pglogical_compat.bc \
+			   compat18/pglogical_compat.o compat18/pglogical_compat.bc \
 			   pglogical_create_subscriber.o
 
 # The # in #define is taken as a comment, per https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=142043

--- a/compat18/pglogical_compat.c
+++ b/compat18/pglogical_compat.c
@@ -1,0 +1,12 @@
+/*-------------------------------------------------------------------------
+ *
+ * pglogical_compat.c
+ *              compatibility functions (mainly with different PG versions)
+ *
+ * Copyright (c) 2015, PostgreSQL Global Development Group
+ *
+ * IDENTIFICATION
+ *              pglogical_compat.c
+ *
+ *-------------------------------------------------------------------------
+ */

--- a/compat18/pglogical_compat.h
+++ b/compat18/pglogical_compat.h
@@ -1,0 +1,172 @@
+#ifndef PG_LOGICAL_COMPAT_H
+#define PG_LOGICAL_COMPAT_H
+
+#include "access/amapi.h"
+#include "access/heapam.h"
+#include "access/table.h"
+#include "access/tableam.h"
+#include "utils/varlena.h"
+
+#define WaitLatchOrSocket(latch, wakeEvents, sock, timeout) \
+	WaitLatchOrSocket(latch, wakeEvents, sock, timeout, PG_WAIT_EXTENSION)
+
+#define WaitLatch(latch, wakeEvents, timeout) \
+	WaitLatch(latch, wakeEvents, timeout, PG_WAIT_EXTENSION)
+
+#define GetCurrentIntegerTimestamp() GetCurrentTimestamp()
+
+#define pg_analyze_and_rewrite(parsetree, query_string, paramTypes, numParams) \
+	pg_analyze_and_rewrite_fixedparams(parsetree, query_string, paramTypes, numParams, NULL)
+
+#define CreateCommandTag(raw_parsetree) \
+	CreateCommandTag(raw_parsetree->stmt)
+
+#define ExecAlterExtensionStmt(stmt) \
+	ExecAlterExtensionStmt(NULL, stmt)
+
+#undef ExecEvalExpr
+#define ExecEvalExpr(expr, econtext, isNull, isDone) \
+	((*(expr)->evalfunc) (expr, econtext, isNull))
+
+#define Form_pg_sequence Form_pg_sequence_data
+
+#define InitResultRelInfo(resultRelInfo, resultRelationDesc, resultRelationIndex, instrument_options) \
+	InitResultRelInfo(resultRelInfo, resultRelationDesc, resultRelationIndex, NULL, instrument_options)
+
+#define ExecARUpdateTriggers(estate, relinfo, tupleid, fdw_trigtuple, newslot, recheckIndexes) \
+	ExecARUpdateTriggers(estate, relinfo, NULL, NULL, tupleid, fdw_trigtuple, newslot, recheckIndexes, NULL, false)
+
+#define ExecARInsertTriggers(estate, relinfo, slot, recheckIndexes) \
+	ExecARInsertTriggers(estate, relinfo, slot, recheckIndexes, NULL)
+
+#define ExecARDeleteTriggers(estate, relinfo, tupleid, fdw_trigtuple) \
+	ExecARDeleteTriggers(estate, relinfo, tupleid, fdw_trigtuple, NULL, false)
+
+#define makeDefElem(name, arg) makeDefElem(name, arg, -1)
+
+#define PGLstandard_ProcessUtility(pstmt, queryString, readOnlyTree, context, params, queryEnv, dest, sentToRemote, qc) \
+	standard_ProcessUtility(pstmt, queryString, readOnlyTree, context, params, queryEnv, dest, qc)
+
+#define PGLnext_ProcessUtility_hook(pstmt, queryString, readOnlyTree, context, params, queryEnv, dest, sentToRemote, qc) \
+	next_ProcessUtility_hook(pstmt, queryString, readOnlyTree, context, params, queryEnv, dest, qc)
+
+#define PGLCreateTrigger(stmt, queryString, relOid, refRelOid, constraintOid, indexOid, isInternal) \
+	CreateTrigger(stmt, queryString, relOid, refRelOid, constraintOid, indexOid, InvalidOid, InvalidOid, NULL, isInternal, false);
+
+#define	PGLDoCopy(stmt, queryString, processed) \
+	do \
+	{ \
+		ParseState* pstate = make_parsestate(NULL); \
+		DoCopy(pstate, stmt, -1, 0, processed); \
+		free_parsestate(pstate); \
+	} while (false);
+
+#define PGLReplicationSlotCreate(name, db_specific, persistency) ReplicationSlotCreate(name, db_specific, persistency)
+
+#ifndef rbtxn_has_catalog_changes
+#define rbtxn_has_catalog_changes(txn) (txn->has_catalog_changes)
+#endif
+
+/* ad7dbee368a */
+#define ExecInitExtraTupleSlot(estate) \
+	ExecInitExtraTupleSlot(estate, NULL, &TTSOpsHeapTuple)
+
+#define ACL_OBJECT_RELATION OBJECT_TABLE
+#define ACL_OBJECT_SEQUENCE OBJECT_SEQUENCE
+
+#define DatumGetJsonb DatumGetJsonbP
+
+#define pgl_heap_attisnull(tup, attnum, tupledesc) \
+	heap_attisnull(tup, attnum, tupledesc)
+
+/* 2a10fdc4307a667883f7a3369cb93a721ade9680 */
+#define getObjectDescription(object) getObjectDescription(object, false)
+
+/* e997a0c642860a96df0151cbeccfecbdf0450d08 */
+#define GetFlushRecPtr() GetFlushRecPtr(NULL)
+
+/* 216a784829c2c5f03ab0c43e009126cbb819e9b2 */
+#define PGLreplorigin_session_setup(node) replorigin_session_setup(node, 0)
+
+/* 19d8e2308bc51ec4ab993ce90077342c915dd116 */
+#define ExecInsertIndexTuples(resultRelInfo, slot, estate, update, noDupErr, specConflict, arbiterIndexes) \
+	ExecInsertIndexTuples(resultRelInfo, slot, estate, update, noDupErr, specConflict, arbiterIndexes, false)
+
+/* 70b42f2790292cc30aa07563f343f7ba6749af01 */
+#define EvalPlanQualInit(epqstate, parentestate, subplan, auxrowmarks, epqParam) \
+	EvalPlanQualInit(epqstate, parentestate, subplan, auxrowmarks, epqParam, NIL)
+
+/* 6a72c42fd5af7ada49584694f543eb06dddb4a87 */
+#define MemoryContextResetAndDeleteChildren(ctx) MemoryContextReset(ctx)
+
+/* 75680c3d805e2323cd437ac567f0677fdfc7b680 */
+#define SPI_push() ((void) 0)
+#define SPI_pop()  ((void) 0)
+#define tuplestore_donestoring(state)  ((void) 0)
+
+/* 89e5ef7e21812916c9cf9fcf56e45f0f74034656 */
+typedef enum ObjectClass
+{
+   OCLASS_CLASS,               /* pg_class */
+   OCLASS_PROC,                /* pg_proc */
+   OCLASS_TYPE,                /* pg_type */
+   OCLASS_CAST,                /* pg_cast */
+   OCLASS_COLLATION,           /* pg_collation */
+   OCLASS_CONSTRAINT,          /* pg_constraint */
+   OCLASS_CONVERSION,          /* pg_conversion */
+   OCLASS_DEFAULT,             /* pg_attrdef */
+   OCLASS_LANGUAGE,            /* pg_language */
+   OCLASS_LARGEOBJECT,         /* pg_largeobject */
+   OCLASS_OPERATOR,            /* pg_operator */
+   OCLASS_OPCLASS,             /* pg_opclass */
+   OCLASS_OPFAMILY,            /* pg_opfamily */
+   OCLASS_AM,                  /* pg_am */
+   OCLASS_AMOP,                /* pg_amop */
+   OCLASS_AMPROC,              /* pg_amproc */
+   OCLASS_REWRITE,             /* pg_rewrite */
+   OCLASS_TRIGGER,             /* pg_trigger */
+   OCLASS_SCHEMA,              /* pg_namespace */
+   OCLASS_STATISTIC_EXT,       /* pg_statistic_ext */
+   OCLASS_TSPARSER,            /* pg_ts_parser */
+   OCLASS_TSDICT,              /* pg_ts_dict */
+   OCLASS_TSTEMPLATE,          /* pg_ts_template */
+   OCLASS_TSCONFIG,            /* pg_ts_config */
+   OCLASS_ROLE,                /* pg_authid */
+   OCLASS_ROLE_MEMBERSHIP,     /* pg_auth_members */
+   OCLASS_DATABASE,            /* pg_database */
+   OCLASS_TBLSPACE,            /* pg_tablespace */
+   OCLASS_FDW,                 /* pg_foreign_data_wrapper */
+   OCLASS_FOREIGN_SERVER,      /* pg_foreign_server */
+   OCLASS_USER_MAPPING,        /* pg_user_mapping */
+   OCLASS_DEFACL,              /* pg_default_acl */
+   OCLASS_EXTENSION,           /* pg_extension */
+   OCLASS_EVENT_TRIGGER,       /* pg_event_trigger */
+   OCLASS_PARAMETER_ACL,       /* pg_parameter_acl */
+   OCLASS_POLICY,              /* pg_policy */
+   OCLASS_PUBLICATION,         /* pg_publication */
+   OCLASS_PUBLICATION_NAMESPACE,   /* pg_publication_namespace */
+   OCLASS_PUBLICATION_REL,     /* pg_publication_rel */
+   OCLASS_SUBSCRIPTION,        /* pg_subscription */
+   OCLASS_TRANSFORM            /* pg_transform */
+} ObjectClass;
+
+#define LAST_OCLASS        OCLASS_TRANSFORM
+
+/* 0fbceae841cb5a31b13d3f284ac8fdd19822eceb added "instrument" param */
+#define index_beginscan(heapRelation, indexRelation, snapshot, nkeys, norderbys) \
+	index_beginscan(heapRelation, indexRelation, snapshot, NULL, nkeys, norderbys)
+
+/*
+ * 27c7c11366f72b1933e298481954a24c742036de added is_merge_update, which is
+ * false for us just like in core execReplication.c.
+ */
+#define ExecBRDeleteTriggers(estate, epqstate, relinfo, tupleid, fdw_trigtuple) \
+	ExecBRDeleteTriggers(estate, epqstate, relinfo, tupleid, fdw_trigtuple, NULL, NULL, NULL, false)
+#define ExecBRUpdateTriggers(estate, epqstate, relinfo, tupleid, fdw_trigtuple, slot) \
+	ExecBRUpdateTriggers(estate, epqstate, relinfo, tupleid, fdw_trigtuple, slot, NULL, NULL, false)
+
+/* cbc127917e04a978a788b8bc9d35a70244396d5b added unpruned_relids */
+#define ExecInitRangeTable(estate, rangeTable, permInfos) \
+	ExecInitRangeTable(estate, rangeTable, permInfos, bms_make_singleton(1))
+
+#endif

--- a/docs/README.md
+++ b/docs/README.md
@@ -90,6 +90,7 @@ If you don’t have PostgreSQL already:
     - PostgreSQL 15: `yum install postgresql15-server postgresql15-contrib`
     - PostgreSQL 16: `yum install postgresql16-server postgresql16-contrib`
     - PostgreSQL 17: `yum install postgresql17-server postgresql17-contrib`
+    - PostgreSQL 18: `yum install postgresql18-server postgresql18-contrib`
 
 ##### Installation
 
@@ -105,6 +106,7 @@ You can proceed to install pglogical for your PostgreSQL version:
  - PostgreSQL 15: `yum install pglogical_15`
  - PostgreSQL 16: `yum install pglogical_16`
  - PostgreSQL 17: `yum install pglogical_17`
+ - PostgreSQL 18: `yum install pglogical_18`
 
 #### Installing pglogical with APT
 
@@ -125,6 +127,7 @@ Debian (e.g. Ubuntu).
     - PostgreSQL 15: `sudo apt-get install postgresql-15`
     - PostgreSQL 16: `sudo apt-get install postgresql-16`
     - PostgreSQL 17: `sudo apt-get install postgresql-17`
+    - PostgreSQL 18: `sudo apt-get install postgresql-18`
 
 ##### Installation
 
@@ -140,6 +143,7 @@ Once pre-requisites are complete, installing pglogical is simply a matter of exe
  - PostgreSQL 15: `sudo apt-get install postgresql-15-pglogical`
  - PostgreSQL 16: `sudo apt-get install postgresql-16-pglogical`
  - PostgreSQL 17: `sudo apt-get install postgresql-17-pglogical`
+ - PostgreSQL 18: `sudo apt-get install postgresql-18-pglogical`
 
 ### From source code
 

--- a/expected/init.out
+++ b/expected/init.out
@@ -54,11 +54,27 @@ BEGIN
 END;
 $$;
 ALTER EXTENSION pglogical UPDATE;
-\dx pglogical
-                   List of installed extensions
-   Name    | Version |  Schema   |          Description           
------------+---------+-----------+--------------------------------
- pglogical | 2.4.5   | pglogical | PostgreSQL Logical Replication
+-- v18 \dx pglogical (v18 added column "Default version")
+SELECT
+	e.extname AS "Name",
+	e.extversion AS "Version",
+	ae.default_version AS "Default version",
+	n.nspname AS "Schema",
+	d.description AS "Description"
+FROM
+	pg_catalog.pg_extension e
+	LEFT JOIN pg_catalog.pg_namespace n ON n.oid = e.extnamespace
+	LEFT JOIN pg_catalog.pg_description d ON
+		d.objoid = e.oid AND
+		d.classoid = 'pg_catalog.pg_extension'::pg_catalog.regclass
+	LEFT JOIN pg_catalog.pg_available_extensions()
+		ae(name, default_version, comment) ON ae.name = e.extname
+WHERE
+	e.extname = 'pglogical' COLLATE pg_catalog.default
+ORDER BY 1;
+   Name    | Version | Default version |  Schema   |          Description           
+-----------+---------+-----------------+-----------+--------------------------------
+ pglogical | 2.4.5   | 2.4.5           | pglogical | PostgreSQL Logical Replication
 (1 row)
 
 SELECT * FROM pglogical.create_node(node_name := 'test_provider', dsn := (SELECT provider_dsn FROM pglogical_regress_variables()) || ' user=super');

--- a/pglogical_apply.c
+++ b/pglogical_apply.c
@@ -135,7 +135,7 @@ typedef struct PGLFlushPosition
 	XLogRecPtr remote_end;
 } PGLFlushPosition;
 
-dlist_head lsn_mapping = DLIST_STATIC_INIT(lsn_mapping);
+static dlist_head lsn_mapping = DLIST_STATIC_INIT(lsn_mapping);
 
 typedef struct ApplyExecState
 {
@@ -152,7 +152,7 @@ struct ActionErrCallbackArg
 	bool is_ddl_or_drop;
 };
 
-struct ActionErrCallbackArg errcallback_arg;
+static struct ActionErrCallbackArg errcallback_arg;
 static TransactionId remote_xid;
 
 static void multi_insert_finish(void);

--- a/pglogical_output_config.c
+++ b/pglogical_output_config.c
@@ -48,7 +48,7 @@ static int32 parse_param_int32(DefElem *elem);
 static void
 process_parameters_v1(List *options, PGLogicalOutputData *data);
 
-enum {
+enum OutputPluginParamKey {
 	PARAM_UNRECOGNISED,
 	PARAM_MAX_PROTOCOL_VERSION,
 	PARAM_MIN_PROTOCOL_VERSION,
@@ -70,7 +70,7 @@ enum {
 	PARAM_HOOKS_SETUP_FUNCTION,
 	PARAM_PG_VERSION,
 	PARAM_NO_TXINFO
-} OutputPluginParamKey;
+};
 
 typedef struct {
 	const char * const paramname;

--- a/sql/init.sql
+++ b/sql/init.sql
@@ -61,7 +61,24 @@ END;
 $$;
 ALTER EXTENSION pglogical UPDATE;
 
-\dx pglogical
+-- v18 \dx pglogical (v18 added column "Default version")
+SELECT
+	e.extname AS "Name",
+	e.extversion AS "Version",
+	ae.default_version AS "Default version",
+	n.nspname AS "Schema",
+	d.description AS "Description"
+FROM
+	pg_catalog.pg_extension e
+	LEFT JOIN pg_catalog.pg_namespace n ON n.oid = e.extnamespace
+	LEFT JOIN pg_catalog.pg_description d ON
+		d.objoid = e.oid AND
+		d.classoid = 'pg_catalog.pg_extension'::pg_catalog.regclass
+	LEFT JOIN pg_catalog.pg_available_extensions()
+		ae(name, default_version, comment) ON ae.name = e.extname
+WHERE
+	e.extname = 'pglogical' COLLATE pg_catalog.default
+ORDER BY 1;
 
 SELECT * FROM pglogical.create_node(node_name := 'test_provider', dsn := (SELECT provider_dsn FROM pglogical_regress_variables()) || ' user=super');
 

